connexion/middleware/main.py

```diff
@@ -0,0 +1,61 @@
+import pathlib
+import typing as t
+
+from starlette.types import ASGIApp, Receive, Scope, Send
+
+
+class ConnexionMiddleware:
+
+    default_middlewares = [
+    ]
+
+    def __init__(
+            self,
+            app: ASGIApp,
+            middlewares: t.Optional[t.List[t.Type[ASGIApp]]] = None
+    ):
+        """High level Connexion middleware that manages a list o middlewares wrapped around an
+        application.
+
+        :param app: App to wrap middleware around.
+        :param middlewares: List of middlewares to wrap around app. The list should be ordered
+                            from outer to inner middleware.
+        """
+        if middlewares is None:
+            middlewares = self.default_middlewares
+        self.app, self.apps = self._apply_middlewares(app, middlewares)
+
+        self._routing_middleware = None
+
+    @staticmethod
+    def _apply_middlewares(app: ASGIApp, middlewares: t.List[t.Type[ASGIApp]]) \
+            -> t.Tuple[ASGIApp, t.Iterable[ASGIApp]]:
+        """Apply all middlewares to the provided app.
+
+        :param app: App to wrap in middlewares.
+        :param middlewares: List of middlewares to wrap around app. The list should be ordered
+                            from outer to inner middleware.
+
+        :return: App with all middlewares applied.
+        """
+        apps = []
+        for middleware in reversed(middlewares):
+            app = middleware(app)
```

Comment:
Indeed, currently `functools.partial` is probably the best way. Open to better options, but it might be hard without tying the implementation to specific middleware types.

I don't think we need to solve this now though. Once we get the internals working, we can still spend some time on the API we want to offer.

Reply:
Yes, just wanted to raise it to be aware of if there was a straightforward solution for it.